_posts/2021-01-12-updating-foreman-inventory-with-system-facts.md

```diff
@@ -0,0 +1,323 @@
+    </tr>
+    <tr>
+      <td>Ignore interfaces with matching identifier</td>
+      <td>Provisioning</td>
+      <td>This is similar to excluded pattern settings above, it's only limited to parsing network interfaces facts. Parsing NICs is more performance heavy operation, therefore has its own configuration. But in general, it can be superceeded by the mentioned setting.</td>
+    </tr>
+    <tr>
+      <td>Ignore Puppet facts for provisioning</td>
+      <td>Provisioning</td>
+      <td>Completely ignore networking information importing from facts, this is useful when the detection does not correctly work in your environment, e.g. you use custom names for bonds or vlan interfaces.</td>
+    </tr>
+  </tbody>
+</table>
+
+### Configuring the Ansible callback
+
+If you're an Ansible user, there is an easy way to keep your Foreman inventory up to date thanks to the foreman.py callback.
+It is also an elegant way to populate (import) unmanaged hosts to your Foreman instance. If you use the installer and install
+[foreman_ansible plugin](https://theforeman.org/plugins/foreman_ansible/), the callback is deployed for any Ansible run
+triggered from Foreman.
+
+However we'll look at how to configure the callback manually into your existing Ansible directory. That way, if you run
+any Ansible playbook, Ansible will upload facts and report to your Foreman. If one or more hosts that the Ansible run
+included does not exist in your Foreman instance, it will be created. If it exists, it will be updated based on the facts
+gathered during the Ansible run. You still need the foreman_ansible to be installed to your Foreman, so it can properly import and process Ansible facts.
+
+Ideally you deploy the Smart Proxy with Ansible feature to the host from which you run Ansible but it's not necessary
+and we'll not rely on this. Just keep that in mind in case you decide to start using the foreman_ansible plugin for running the Ansible.
+
+Start by installing the foreman_ansible plugin. To do that, run the installer like this
+
+```
+foreman-installer --enable-foreman-plugin-ansible
+```
+
+Don’t worry if you ran installer with other arguments before, they are stored and automatically added to the next installer run. If you plan to use Ansible with Foreman further, it is a good idea to also install Smart Proxy plugin and CLI plugin, to do so, run
+
+```
+foreman-installer --enable-foreman-plugin-ansible \
+                  --enable-foreman-proxy-plugin-ansible \
+                  --enable-foreman-cli-ansible
+```
+
+Then proceed to the Ansible configuration. First of all, you need to download the callback. It lives in the [Foreman Ansible Module](https://galaxy.ansible.com/theforeman/foreman)
+collection.
+Just run the following command  on the machine on which you normally run `ansible-playbook` on.
```

Comment:
Having read part way down, I find myself asking should this be some server I already have that is outside Foreman's knowledge or is this my Foreman host or is this an external Smart Proxy. I can see value in doing two things here:

 1) Pushing a recommended, easy happy path (green field approach)
 2) Calling out the brown field approach where you already have some Ansible setup, and you want to start populating Foreman (should we recommend enrolling the control node in Foreman to make certificates easier? or installing a smart proxy?)

Reply:
this was meant for the brown field case, I added clarification for both cases though